### ISSUE TITLE
Accidental change of map rotation

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#16751]
I believe that changing the dead zone for initiating the manual map rotation from 30 deg to just 5 deg here https://github.com/osmandapp/OsmAnd/pull/16746 increases the risk of accidentally  changing to manual map rotation when just pinch-zooming. Manual rotation is actually a rather rare use case for the majority of users...

**Comment by sonora** [comment]
Number 3 I don't understand. How can we prevent accidental map rotation if we do not have a threshold? Please note the threshold is **only to activate** the map rotation, not to limit the rotation itself. (I.e. turning 30 degrees clockwise then without interruption 27 degrees back does of course work.)

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14261mqta, Released: 2023-03-23
Map zooming causes random spins


https://user-images.githubusercontent.com/104760013/227206437-0a8577e8-3ceb-4f73-9275-2fc2699f52ee.mp4

**Comment by sonora** [comment]
I guess of we don'tlike the dead zone of having to twist at least 30 degrees to change the map from "North is up" to "Manual rotation" (immediately after that you can rotate to anything,l, maybe we should **disable twisting entirely** in the "North is up" mode. That would result in users who need the comparatively rarely used "Manual rotation" mode would have to explicitly tap the compass icon or use  the settings to explicitly switch to manual rotation.... and it would protect the vast majority of users who need "North is up" and no accidental switching...

**Comment by pebogufi** [comment]
There was a change in the last days. 
Now it is confusing, but before that change it was ok.
What was the change exactely, can we return to the old behaviour ?

**Comment by sonora** [comment]
I think it started by trying to have pinch zoom and rotating the map combined  into one single gesture. But does anybody have a use case to perform rhese two rather different things in one go? Not me...

**Comment by dmpr0** [comment]
> There was a change in the last days. Now it is confusing, but before that change it was ok. What was the change exactely, can we return to the old behaviour ?

Could you please provide details about confusing changes?

**Comment by pebogufi** [comment]
For me it is irritating if I only want to make a pinch zoom, when the map rotates accidentally and I have to reset the direction each time. 
(I am human using fingers)

**Comment by pebogufi** [comment]
See also https://github.com/osmandapp/OsmAnd/issues/16805

**Comment by sonora** [comment]
@dmpr0  Yeah 😉 , but I thnk in the vast majority of everyday map use users want to either zoom or rotate, but not both in one go?

If you don't like the 30 degrees threshold solution (we had it for almost 10 years!), an alternative is
- we disable rotation in "North is up" mode entirely (which would suite most users just fine),
- and expect those few users who need parallel zoom+rotate to explicitly switch to the "Manual rotation" mode (that's what it's there for)?

**Comment by Boothy99** [comment]
> ....maybe we should **disable twisting entirely** in the "North is up" mode. That would result in users who need the comparatively rarely used "Manual rotation" mode would have to explicitly tap the compass icon or use the settings to explicitly switch to manual rotation.... and it would protect the vast majority of users who need "North is up" and no accidental switching...

I agree strongly with you @sonora - "North ALWAYS up (fixed)" is exactly what most folk appear to expect by the looks of the multiple issues posted relating to this. Yes, "fixed" should be just that - "fixed", and if in the rare occasion someone wants "rotated", then an icon tap should suffice to execute it.

**Comment by dmpr0** [comment]
> @dmpr0 Yeah 😉 , but I thnk in the vast majority of everyday map use users want to either zoom or rotate, but not both in one go?
> 
> If you don't like the 30 degrees threshold solution (we had it for almost 10 years!) to trigger "Manual rotation" mode, an alternative is
> 
> * we disable rotation in "North is up" mode entirely (which would suite most users just fine),
> * and expect those few users who need parallel zoom+rotate to explicitly switch to the "Manual rotation" mode (that's what it's there for)?

Where are from you get statistic?

**Comment by vshcherb** [comment]
As of 27.03 Current build has designed  behavior for all tested behaviors. Last glitch needs to be reviewed
- Zoom in is only possible by initial step of 0.3: we want to have a threshold start 0.1

**Comment by vshcherb** [comment]
>  Yeah 😉 , but I thnk in the vast majority of everyday map use users want to either zoom or rotate, but not both in one go?

We need to stick to the main concept that fingers are linked to the map location, so you always should be able to rotate and zoom in cause your fingers will move around the map. It's a physical paradigm like working with gypsum (sculpture)

**Comment by Boothy99** [comment]
@vshcherb it really is frustrating when one selects North is up (fixed); zooms in with two finger, only to find North is no-longer fixed and the map is now skewed. This, after the very merest accidental and unavoidable "twist" of ones fingers by the smallest of degrees. Please revert this 🤞 I hate looking at a skewed map when all I wanted was a zoomed in / out, Northward facing view.

**Comment by sonora** [comment]
As a side note, @dmpr0: Can I maybe invite you to review your 4 contributions to the above discussion: There is not a single one with any argument regarding the technical suggestions which have been made (trigger threshold vs. limiting rotation to the "Manual profile"), no comment regarding the usability issues several users describe, and no answer to the question why we got into this quagmire in the first place, i.e. what is the use case someone actually tried to fix when breaking this? IMHO, we as the OsmAnd team could do a better job in responding to user issues and discussion improvement suggestions?

**Comment by dmpr0** [comment]
> **We broke it, folks!**
> 
> What's so hard to understand about it? Let's acknowledge and fix it!

So, you want to perform rotate gesture, but don't want the map to follow the gesture?

**Comment by sonora** [comment]
There's 2 suggestions: Keep the "fixed" mode "fixed". Or re-introduce a big enough trigger threshold.

**Comment by sonora** [comment]
Here's the pull request I propose https://github.com/osmandapp/OsmAnd/pull/16830: A 30 degree trigger guard for the "North is up (fixed)" mode, while all other modes remain with the new low trigger limits. I think this should suite everyone?

**Comment by vshcherb** [comment]
> We broke it, folks!
> What's so hard to understand about it? Let's acknowledge and fix it!

Nope it was not correctly implemented, check standard programs Google / Apple Maps as they have much larger audience, so default map display should be the same. And that's how it is right now. 

Other special modes probably could be used by advanced users but they couldn't be standard

**Comment by vshcherb** [comment]
I think This is a discussion rather than the issue
